Add support for IOCTL

steinbeck65 opened the pull request:
ioctl.py - example code for handling IOCTL requests
ioctl.c - example code for sending IOCTL requests

eestrada commented:
@steinbeck65 This looks good to me. Before merging this, will you first move `ioctl.py` and `iotctl.c` to live under `examples`? I am trying to keep the root directory as minimal as possible.

steinbeck65 commented:
I've moved the ioctl.c and ioctl.py files to the examples directory and
squashed those changes into the original commit, hopefully I did that all
correctly. Let me know if I can do anything else.

Thank you very much for breathing new life into fusepy, I and I'm sure many
other really appreciate it!

--
Greg

On Tue, Jan 30, 2018 at 9:53 PM, Ethan Estrada <notifications@github.com>
wrote:

> @steinbeck65 <https://github.com/steinbeck65> This looks good to me.
> Before merging this, will you first move ioctl.py and iotctl.c to live
> under examples? I am trying to keep the root directory as minimal as
> possible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fusepy/fusepy/pull/62#issuecomment-361823531>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AI7IUCx13ResSBigI-VGyHzNxlYx-ghCks5tP_HcgaJpZM4Kt9Rf>
> .
>

eestrada commented:
Thanks for taking care of that. The squashing looks like it worked correctly.

You're welcome! `fusepy` is an important project that needs to remain alive. I'm happy to do what I can to support it.

terencehonles commented on examples/ioctl.py:
```diff
@@ -0,0 +1,20 @@
+from errno import ENOTTY
+from ioctl_opt import IOWR
+
+def ioctl(self, path, cmd, arg, fh, flags, data):
```
Can you define the whole operations class? As it stands this example "does not work"

steinbeck65 commented:
I created a minimal fuse file system using memory.py as a starting point,
then added in the ioctl support, so it now works as a standalone example.

I've squashed these changes into the original pull request, let me know if
that is satisfactory or if you would like any additional changes.

Thanks

--
Greg

On Sun, Feb 11, 2018 at 4:05 PM, Terence Honles <notifications@github.com>
wrote:

> *@terencehonles* commented on this pull request.
>
> Thanks for the contribution 🙂
> ------------------------------
>
> In examples/ioctl.py
> <https://github.com/fusepy/fusepy/pull/62#discussion_r167453907>:
>
> > @@ -0,0 +1,20 @@
> +#!/usr/bin/env python
> +# Example code for handling IOCTLs in FUSE client
> +
> +import ctypes
> +import struct
> +from errno import ENOTTY
> +from ioctl_opt import IOWR
> +
> +def ioctl(self, path, cmd, arg, fh, flags, data):
>
> Can you define the whole operations class? As it stands this example "does
> not work"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fusepy/fusepy/pull/62#pullrequestreview-95673831>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AI7IUFM2ODdrJBgnaZQHY3Cp7MvvJ8Unks5tT3I1gaJpZM4Kt9Rf>
> .
>
